### PR TITLE
Add info about async query & save

### LINKED ISSUE #126: Saving Data
Required pages:
- [ ] basics
- [ ] related data
- [ ] cascade delete
- [ ] async
- [ ] explicit values into generated columns
- [ ] handling concurrency conflicts

### LINKED ISSUE #158: Querying Data
Needs to cover
- Basic LINQ queries
- AsNoTracking, QueryTrackingBehavior
- Include, ThenInclude
- Need to await async operations
- Raw SQL queries

### PR DESCRIPTION
**Pull request by rowanmiller**
Resolves #158
Resolves #126

### CODE REVIEW

**Comment by bradygaster** [comment]
### :white_check_mark: Validation status: passed


For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C1%5C25%5Cc1705ea6-142e-cb02-f3b5-8dd98c6d1733%5CPullRequest%5C201701251837000458-340%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by rowanmiller** [comment]
cc @ajcvickers @divega @AndriySvyryd could one of you review this.

File: samples/core/Saving/Saving/Async/Sample.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        public static void Run()
+        {
+            using (var db = new BloggingContext())
+            {
+                db.Database.EnsureDeleted();
+                db.Database.EnsureCreated();
+            }
+
+            AddBlogAsync("http://sample.com").Wait();
```
Comment (ajcvickers) [inline samples/core/Saving/Saving/Async/Sample.cs]: Probably not the best idea to show use of .Wait when it is so easy to use this incorrectly and get a deadlock. cc @AndriySvyryd

Comment (rowanmiller) [inline samples/core/Saving/Saving/Async/Sample.cs]: This isn't shown in the docs... it's just to allow the console app to run the async method. Is there a better way to do this (call an async method from a sync entry point)?

Comment (ajcvickers) [inline samples/core/Saving/Saving/Async/Sample.cs]: Probably okay then, but interesting to hear if there is a better way.

Comment (AndriySvyryd) [inline samples/core/Saving/Saving/Async/Sample.cs]: It's acceptable to have exactly one `Wait` in you app: in the `Main` method.

**Comment by bradygaster** [comment]
### :white_check_mark: Validation status: passed


For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C1%5C25%5Cc1705ea6-142e-cb02-f3b5-8dd98c6d1733%5CPullRequest%5C201701252235151631-340%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**